src/DynamoCore/Graph/Workspaces/SerializationConverters.cs

```diff
@@ -230,15 +230,13 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
             else if (typeof(DSFunctionBase).IsAssignableFrom(type))
             {
                 var mangledName = obj["FunctionSignature"].Value<string>();
-                var priorNames = libraryServices.GetPriorNames();
-                var functionDescriptor = libraryServices.GetFunctionDescriptor(mangledName);
-                string newName;
-
-                // Update the function descriptor if a newer migrated version of the node exists
-                if (priorNames.TryGetValue(mangledName, out newName))
+                var hintSignature = libraryServices.FunctionSignatureFromFunctionSignatureHint(mangledName);
```

Comment:
could simplify code here with coalescing operator :
`
var lookupSignature = libraryServices.FunctionSignatureFromFunctionSignatureHint(mangledName) ?? mangledName;
`
Unless your way is easier to read/debug.

Reply:
will update.